adpkd_segmentation/inference/ensemble_utils.py

```diff
@@ -0,0 +1,90 @@
+import os
+from pathlib import Path
+import nibabel as nib
+import numpy as np
+
+
+def get_tail(dir=".", depth=1):
+    """
+    Given a folder path OR directory, return
+    the folder given the depth.
+    Note: depth = 1 denotes the newest folder
+    in the tree.
+    """
+    t_head = dir
+    num_depth = depth
+    if os.path.isfile(dir):
+        num_depth += 1
+    #
+    idx = 0
+    while idx < num_depth:  # test
+        t_head, t_tail = os.path.split(t_head)
+        idx += 1
+    #
+    return t_tail  # Tested, good
+
+
+def get_scan(config, dir="."):
+    """This is designed for the output folder pattern in
+    ADPKD-segmentation-pytorch (Goel et al. Radiology AI). At inference, the
+    nifti files are saved in
+    .../saved_inference/repo_name/patient_ID/scan_folder/ITKSNAP_DCM_NIFTI/pred.vol.nii
+    You can find Dr. Goel's repo at:
+    https://github.com/aksg87/adpkd-segmentation-pytorch.git
+    And you can find his deployment:
+    web: https://pubs.rsna.org/doi/10.1148/ryai.210205
+    doi: https://doi.org/10.1148/ryai.210205
+    """
+    target_depth = 1
+    if get_tail(dir, target_depth) == config["youngest_child"]:
+        target_depth += 1
+    return get_tail(dir, target_depth)
+
+
+def grab_organ_dirs(iter_dict, config):
```

Comment:
Passing in config - dictionary as an input makes the function very hard to define. Let's try to refactor so the inputs are explicit! @Dom-49

Reply:
I've refactored the code and tested everything out. I'll be recommitting within the next 30 minutes or so.